app/models/user.rb

```diff
@@ -738,8 +738,17 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    if SiteSetting.restrict_letter_avatar_colors.present?
```

Comment:
Since we're only change the list of available colors, here's how I would have written it 😉 

``` ruby
colors = SiteSetting.restrict_letter_avatar_colors.present? ? 
  SiteSetting.restrict_letter_avatar_colors.split("|") : 
  LetterAvatar::COLORS
color = select_color(username, colors)
color.map { |c| c.to_s(16).rjust(2, '0') }.join
```

Reply:
`color.map` step is required only when using default colors to turn them into hex, so there is another check for `SiteSetting.restrict_letter_avatar_colors` needed anyway